kitten-scientists.user.js

```diff
@@ -712,25 +734,39 @@ var run = function() {
     ReligionManager.prototype = {
         manager: undefined,
         crafts: undefined,
-        build: function (name) {
-            var build = this.getBuild(name);
-            var button = this.getBuildButton(name);
+        build: function (name, variant) {
+            var build = this.getBuild(name, variant);
+            var button = this.getBuildButton(name, variant);
 
             if (!button || !button.model.enabled) return;
 
             //need to simulate a click so the game updates everything properly
             button.domNode.click(build);
             storeForSummary(name, 1, 'faith');
-
-            activity('Kittens have discovered ' + build.label, 'ks-faith');
-        },
-        getBuild: function (name) {
-            return game.religion.getRU(name);
+			if (variant === "s") {
+				activity('Kittens have discovered ' + build.label, 'ks-faith');
+			} else {
+				activity('Kittens have built a new ' + build.label, 'ks-faith');
+			}
         },
-        getBuildButton: function (name) {
-            var buttons = this.manager.tab.rUpgradeButtons;
-            var build = this.getBuild(name);
-
+		getBuild: function (name, variant) {
+			if (variant === "z"){
+				return game.religion.getZU(name);
+			}else if (variant === "s"){
```

Comment:
I know this isn't completely consistent across the file, however, I prefer we use the following layout for if/else statements:

```javascript
if (condition) {
    // results true
} else {
    // results false
}
```

If there are more than 4 cases, then consider using a switch case instead. In this case I will leave it up to your discretion since there are only 3 cases.